x/slashing/keeper/epoch.go

```diff
@@ -0,0 +1,138 @@
+package keeper
+
+// +TODO just copied from staking module
+// There should be a way to simplify this
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	db "github.com/tendermint/tm-db"
+)
+
+// keys
+var (
+	NextEpochActionID      = []byte("next_epoch_action_id")
+	EpochActionQueuePrefix = "epoch_action"
+)
+
+// SetNextEpochActionID save ID to be used for next epoch action
+func (k Keeper) SetNextEpochActionID(ctx sdk.Context, actionID uint64) {
+	store := ctx.KVStore(k.storeKey)
+
+	store.Set(NextEpochActionID, sdk.Uint64ToBigEndian(actionID))
+}
+
+// GetNextEpochActionID returns ID to be used for next epoch
+func (k Keeper) GetNextEpochActionID(ctx sdk.Context) uint64 {
+	store := ctx.KVStore(k.storeKey)
+
+	bz := store.Get(NextEpochActionID)
+	if bz == nil {
+		// return default action ID to 1
+		return 1
+	}
+
+	return sdk.BigEndianToUint64(bz)
+}
+
+// ActionStoreKey returns action store key from ID
+func ActionStoreKey(epochNumber int64, actionID uint64) []byte {
+	return []byte(fmt.Sprintf("%s_%d_%d", EpochActionQueuePrefix, epochNumber, actionID))
+}
+
+// SaveEpochAction save the actions that need to be executed on next epoch
+func (k Keeper) SaveEpochAction(ctx sdk.Context, epochNumber int64, action sdk.Msg) {
```

Comment:
```suggestion
func (k Keeper) QueueMsgForEpoch(ctx sdk.Context, epochNumber int64, action sdk.Msg) {
```

Revisit if theres a better way to name this